src/coreclr/vm/threadsuspend.cpp

```diff
@@ -4915,7 +4915,9 @@ bool IsSpecialCaseAsyncRet(MethodDesc* pMD)
     // causing loading to happen? Also, can we just mark them as async2 in SPC,
     // or force them to be fully interruptible?
     LPCUTF8 name = pMD->GetName();
-    return strcmp(name, "UnsafeAwaitAwaiterFromRuntimeAsync") == 0 || strcmp(name, "AwaitAwaiterFromRuntimeAsync") == 0;
+    return strcmp(name, "UnsafeAwaitAwaiterFromRuntimeAsync") == 0 ||
+        strcmp(name, "AwaitAwaiterFromRuntimeAsync") == 0 ||
+        strcmp(name, "Await") == 0;
```

Comment:
This can probably be removed now that these are marked as runtime-async via MethodImpl

Reply:
Yes, nothing special about the helpers names now.

I think the whole special-casing of async function in hijacking will be unnecessary as we stop caring about return kinds, except on x86